src/daft-scan/src/lib.rs

```diff
@@ -321,4 +321,12 @@ impl Pushdowns {
             limit: self.limit,
         }
     }
+
+    pub fn with_columns(&self, columns: Option<Arc<Vec<String>>>) -> Self {
```

Comment:
Shouldn't this also be an indexset to avoid recopying all the data for each optimization pass?

Reply:
The `Pushdowns` struct needs to impl `Hash` because it is held by the logical plan node. However, `IndexSet` does not implement `Hash`.

I do think that ideally this should be IndexSet though, lmk if you have any other solutions here